### PR TITLE
WASM 2 JSON

### PR DESCRIPTION
**Pull request by lucksus**
Use `ribosome::dna_run()` (from core) to get JSON from `__hdk_get_json_definition()` from WASM and include that in the output.

This almost works as expected! But not quite. The output is on the wrong level..

To test this, build `hc` this branch and point the Zome's Cargo.toml to a holohchain-rust that has these changes merged in: https://github.com/holochain/holochain-rust/pull/549/files.

You should see the JSON printed on screen and it should appear in the `bundle.json` output.

It is on the wrong level because `fn bundle_recurse(&self, path: &PathBuf) -> DefaultResult<Object>` is recursive (doh!) and the level on which we get the WASM from the build process is to deep in the tree for attaching the whole zome JSON.

### CODE REVIEW

**Comment by lucksus** [comment]
Tried to finish this tonight but really have to sleep now.
 @sphinxc0re, @Connoropolous, @maackle, someone, please fix the tree-level problem 🙏 then we can get rid of redundant JSON code in our DNA source :)

File: src/cli/package.rs [hunk 1]
```diff
@@ -131,6 +137,26 @@ impl Packager {
...
+                            &ZomeFnCall::new(&"", &"", &"__hdk_get_json_definition", &""),
+                            Some(String::from("{}").into_bytes())
+                        );
+                        println!("RESULT: {:?}", result);
+                        result.ok()
+                    }).and_then(|call_result| {
+                        println!("WASM2JSON: {}", call_result);
+                        serde_json::from_str(&call_result).ok()
+                    }).and_then(|json_from_wasm| {
+                        let parent_filename= util::file_name_string(&node.parent().unwrap().to_path_buf()).unwrap();
```
Comment (lucksus) [inline src/cli/package.rs]: This was an attempt to attach it to the level above but it just changes the root's name ;) still on the level of "code", but with "code" replaced by "blog".. :/

**Comment by Connoropolous** [comment]
@lucksus fixed the recursive tree problem :) now the JSON just needs some minor tweaks. It's slightly different than core expects in a DNA file

File: src/cli/package.rs [hunk 2]
```diff
@@ -119,7 +125,48 @@ impl Packager {
...
+                    .read_dir()?
+                    .filter(|e| e.is_ok())
+                    .map(|e| e.unwrap().path())
+                    .filter(|path| {
+                        let maybe_metadata = fs::metadata(path);
+                        match maybe_metadata {
+                            Ok(metadata) => metadata.is_dir(),
+                            Err(_) => false,
+                        }
+                    })
```
Comment (sphinxc0re) [inline src/cli/package.rs]: Why? Just do

```rust
.filter(|path| path.is_dir())
```

File: src/cli/package.rs [hunk 3]
```diff
@@ -119,7 +125,48 @@ impl Packager {
...
+
+                    let wasm_binary = base64::decode(&wasm)?;
+
+                    let context= test_context("HC");
+                    let call_result = ribosome::run_dna(
+                        &"HC",
+                        context,
+                        wasm_binary,
+                        &ZomeFnCall::new(&"", &"", &"__hdk_get_json_definition", &""),
+                        Some(String::from("{}").into_bytes())
```
Comment (sphinxc0re) [inline src/cli/package.rs]:
```suggestion
                        Some("{}".as_bytes().to_vec())
```
So we don't have to allocate a complete string for this simple manoeuvre.

File: src/cli/test_context.rs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+use holochain_core::{
+    context::Context,
+    logger::Logger,
+    persister::SimplePersister,
+};
+use holochain_agent::Agent;
+use holochain_cas_implementations::{cas::file::FilesystemStorage, eav::file::EavFileStorage};
+use std::sync::{Arc, Mutex};
+
+
```
Comment (thedavidmeister) [inline src/cli/test_context.rs]:
```suggestion
```

**Pull request merged**